tests/handlers/aws/test_logstash_output.py

```diff
@@ -0,0 +1,288 @@
+            """
+        lgc.with_env("CONFIG_STRING", logstash_config)
+        lgc.with_command(
+            'bash -c "/opt/logstash/bin/logstash-plugin install '
+            'logstash-input-elastic_serverless_forwarder && /opt/logstash/bin/logstash"'
+        )
+        self.logstash = lgc.start()
+
+        self.fixtures = {
+            "cw_log_1": _load_file_fixture("cloudwatch-log-1.json"),
+            "cw_log_2": _load_file_fixture("cloudwatch-log-2.json"),
+        }
+
+        group_name = _class_based_id(self, suffix="source-group")
+        stream_name = _class_based_id(self, suffix="source-stream")
+
+        _logs_create_cloudwatch_logs_group(self.logs_client, group_name=group_name)
+        cw_logstream = _logs_create_cloudwatch_logs_stream(
+            self.logs_client, group_name=group_name, stream_name=stream_name
+        )
+        self.cloudwatch_group_arn = cw_logstream["arn"]
+
+        _logs_upload_event_to_cloudwatch_logs(
+            self.logs_client,
+            group_name=group_name,
+            stream_name=stream_name,
+            messages_body=[self.fixtures["cw_log_1"] + self.fixtures["cw_log_2"]],
+        )
+
+        self.group_name = group_name
+        self.stream_name = stream_name
+
+        os.environ["AWS_DEFAULT_REGION"] = aws_default_region
+        os.environ["SQS_CONTINUE_URL"] = _sqs_create_queue(self.sqs_client, _class_based_id(self, suffix="-continuing"))
+        os.environ["SQS_REPLAY_URL"] = _sqs_create_queue(self.sqs_client, _class_based_id(self, suffix="-replay"))
+
+        self.mocks = {
+            "s3client": mock.patch("storage.S3Storage._s3_client", new=self.s3_client),
+            "cloudwatchclient": mock.patch("handlers.aws.utils.get_cloudwatch_logs_client", lambda: self.logs_client),
+            "sqsclient": mock.patch("handlers.aws.utils.get_sqs_client", lambda: self.sqs_client),
+            "sqsclient2": mock.patch("handlers.aws.handler.get_sqs_client", lambda: self.sqs_client),
+        }
+        for k, m in self.mocks.items():
+            m.start()
+
+    def tearDown(self) -> None:
+        self.localstack.stop()
+        self.logstash.stop()
+
+        for k, m in self.mocks.items():
+            m.stop()
+
+    def test_sent_messages(self) -> None:
+        os.environ["S3_CONFIG_FILE"] = _prepare_config_file(
+            self,
+            "config.yaml",
+            dict(CloudwatchLogStreamARN=self.cloudwatch_group_arn, LogstashURL=self.logstash.get_url()),
+            "folder/config.yaml",
+        )
+
+        event_cloudwatch_logs, event_ids_cloudwatch_logs = _logs_retrieve_event_from_cloudwatch_logs(
+            self.logs_client, group_name=self.group_name, stream_name=self.stream_name
+        )
+
+        ctx = ContextMock(TIMEOUT_15m)
+        handler(event_cloudwatch_logs, ctx)  # type: ignore
+
+        msgs = self.logstash.get_messages()
+        assert len(msgs) == 2
+
+        assert msgs[0]["message"] == self.fixtures["cw_log_1"].rstrip("\n")
```

Comment:
I'd assert everything we expect in `msg[]`, like we do [here](https://github.com/elastic/elastic-serverless-forwarder/pull/210/files#diff-405288248598cf246486803f67aaf3389026f2e92321efdfa325efe6c42094daR268-R277)